Use upstream gcloud-sdk image instead of our oudated one

legoktm opened the pull request:
## Status

Ready for review

## Description of Changes

We have not updated our gcloud-sdk image since 2018 and it's really not worth us maintaining this when Google provides its own image.

We now authenticate using the method documented at <https://cloud.google.com/sdk/docs/downloads-docker>, in which we invoke the container to login, which creates a volume that can be used in future calls. This means we no longer need to have a running container that we exec into; we can just create a new container each time we want to make a gcloud call.

Fixes #7195.
Fixes #7196.

## Testing

How should the reviewer test this PR?

* [ ] CI, especially staging job, passes
* [ ] Visual review

## Deployment

Any special considerations for deployment? n/a

## Checklist

- [x] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass
- [x] I have written a test plan and validated it for this PR
- [x] These changes do not require documentation

legoktm committed:
Use upstream gcloud-sdk image instead of our oudated one

We have not updated our gcloud-sdk image since 2018 and it's really not
worth us maintaining this when Google provides its own image.

We now authenticate using the method documented at
<https://cloud.google.com/sdk/docs/downloads-docker>, in which we invoke
the container to login, which creates a volume that can be used in
future calls. This means we no longer need to have a running container
that we exec into; we can just create a new container each time we want
to make a gcloud call.

Fixes #7195.
Fixes #7196.

cfm commented on devops/gce-nested/ci-env.sh:
```diff
@@ -48,26 +48,25 @@ function generate_gce_creds_file() {
         fi
     fi
+
+    if ! docker inspect --format '{{ .Id }}' gcloud-config > /dev/null 2>&1; then
```
```suggestion
    # Prime the "gcloud-config" container by authenticating from $GCE_CREDS_FILE.
    if ! docker inspect --format '{{ .Id }}' gcloud-config > /dev/null 2>&1; then
```

cfm commented on devops/gce-nested/ci-env.sh:
```diff
@@ -48,26 +48,25 @@ function generate_gce_creds_file() {
+    fi
 }
 
 # Wrapper function to communicate with the gcloud API. Ensure gcloud-sdk
 # container is running, and if so, pass all args to it.
```
```suggestion
# Each gcloud_call() invocation just runs a fresh (note "--rm") instance of the "google-cloud-cli" container primed from "gcloud-config" above.
```

cfm reviewed (changes requested):
This is a nice improvement both operationally and conceptually.  Because it changes the conceptual model of this CI workflow, I think it's worth documenting further, so I've proposed two comments inline.

legoktm committed:
Improve documentation in gcloud functions

Co-authored-by: Cory Francis Myers <cory@freedom.press>

legoktm commented:
Thanks, accepted both.

legoktm committed:
Try using `virtualenv` for venv creation if possible

In e0356370bc4f3 I flipped this to use `python -m venv` but some
environments (staging) don't have that, so let's restore the old
behavior of using `virtualenv`, but fall back to `python -m venv`
if the former fails.

cfm reviewed (approved):
Great.  Thanks, @legoktm.
